pcdsdevices/analog_signals.py

```diff
@@ -0,0 +1,167 @@
+import time
+import pcdsdevices.utils as key_press
+from ophyd import Device, Component as Cpt, EpicsSignal, EpicsSignalRO
+
+
+class Acromag(Device):
+    """
+    Class for Acromag analog input/ouput signals
+
+    Parameters:
+    -----------
+    prefix : str
+        The Epics base of the acromag
+
+    name : str
+        A name to prefer to the device
+    """
+    # Components for each channel
+    ao1_0 = Cpt(EpicsSignal, ":ao1:0", kind='normal')
```

Comment:
This is a good use case for a `DynamicDeviceSignal`:

```python
from ophyd import DynamicDeviceComponent as DDC, Device

def _make_signals(channels):
    return {
        'signal{}'.format(channel): (EpicsSignal, ':ai1:{}'.format(channel),
                                     dict(kind='normal', write_pv=':ao1:{}'.format(channel)))
        for channel in channels

    }

class Acromag(Device):
    signals = DDC(_make_signals(range(0, 16)))
```

You could then access it by:

```python
In [1]: dev = Acromag('prefix', name='dev')

In [2]: dev.signals.signal0.pvname
Out[2]: 'prefix:ai1:0'

In [3]: dev.signals.signal0.setpoint_pvname
Out[3]: 'prefix:ao1:0'

In [4]: dev.signals.component_names
Out[4]:
('signal0',
 'signal1',
 'signal2',
 'signal3',
 'signal4',
 'signal5',
 'signal6',
 'signal7',
 'signal8',
 'signal9',
 'signal10',
 'signal11',
 'signal12',
 'signal13',
 'signal14',
 'signal15')
```

Reply:
I like this, but it isn't quite correct. The in and the out signals are not correlated (i.e they should not be in the same `EpicsSignal` as read and write pairs).